fgpyo/samtools.py

```diff
@@ -0,0 +1,375 @@
+        args.append("--reverse-complement")
+    if fastq:
+        args.append("--fastq")
+
+    if fai_idx is not None:
+        args.extend(["--fai-idx", str(fai_idx)])
+
+    if gzi_idx is not None:
+        args.extend(["--gzi-idx", str(gzi_idx)])
+
+    if region_file is not None:
+        args.extend(["--region-file", str(region_file)])
+
+    args.append(str(sequence_file))
+
+    if regions is not None:
+        args.extend(regions)
+
+    return _pysam_faidx(*args)
+
+
+@enum.unique
+class SamIndexType(enum.Enum):
+    BAI = "-b"
+    CSI = "-c"
+
+
+def index(
+    alignment_file: Union[Path, Sequence[Path]],
+    output: Optional[Path] = None,
+    threads: int = 0,
+    index_type: SamIndexType = SamIndexType.BAI,
+    csi_index_size: int = 14,
+) -> str:
+    """
+    Calls the `samtools` function `index` using the pysam samtools dispatcher.
+
+    Arguments will be formatted into the appropriate command-line call which will be invoked
+    using the pysam dispatcher.
+
+    Returns the stdout of the resulting call to pysam.index()
+
+    Args:
+        alignment_file: Path to the SAM/BAM/CRAM file to index.
+        output: Path to the file to write output. (Currently may only be used when exactly one
+            alignment file is being indexed.)
+        threads: Number of input/output compression threads to use in addition to main thread.
+        index_type: The type of index file to produce when indexing.
+        csi_index_size: Sets the minimum interval size of CSI indices to 2^INT.
+    """
+
+    if isinstance(alignment_file, Path):
+        alignment_file = [alignment_file]
+    if output is not None and len(alignment_file) != 1:
+        raise ValueError("Only one output may be specified, for indexing only one alignment file")
+
+    args = [] if len(alignment_file) == 1 else ["-M"]
+
+    if index_type != SamIndexType.BAI:
+        args.append(index_type.value)
+
+    if index_type == SamIndexType.CSI:
+        args.extend(["-m", str(csi_index_size)])
+    args.extend(["-@", str(threads)])
+    args.extend(f"{input_path}" for input_path in alignment_file)
+    if output is not None:
+        args.extend(["-o", str(output)])
+
+    return _pysam_index(*args)
+
+
+def sort(
+    alignment_file: Path,
+    output: Path,
+    index_output: bool = True,
+    sort_unmapped_reads: bool = False,
+    kmer_size: int = 20,
+    compression_level: Optional[int] = None,
+    memory_per_thread: str = "768MB",
+    sort_order: SamOrder = SamOrder.Coordinate,
+    sort_tag: Optional[str] = None,
+    output_format: SamFileType = SamFileType.BAM,
+    tempfile_prefix: Optional[str] = None,
+    threads: int = 1,
+    no_pg: bool = False,
+) -> str:
+    """
+    Calls the `samtools` function sort using the pysam samtools dispatcher.
+
+    Arguments will be formatted into the appropriate command-line call which will be invoked
+    using the pysam dispatcher.
+
+    Returns the stdout of the resulting call to pysam.sort()
```

Comment:
For Google style docstrings, use the `Returns:` block like you would an `Args:` or `Attributes`.

Only making this comment here, but it applies to all other docstrings too.

https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html